vendmachine/VendingMachine.java

```diff
@@ -0,0 +1,130 @@
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.io.FileWriter;
+import java.io.FileReader;
+import java.io.BufferedWriter;
+import java.io.BufferedReader;
+import java.io.IOException;
+
+class VendingMachine {
+    private HashMap<String, Product> products = new HashMap<>();
+    private BufferedWriter log = null;
+    private String user = null;
+
+    private void close() {
+        try {
+            BufferedWriter state = new BufferedWriter(new FileWriter("state.csv"));
```

Comment:
💡 I would suggest super class here as best practice: `Writer` instead of `BufferedWriter`.